### PR TITLE
Removes outputStyle option

### PR DESCRIPTION
**Pull request by xcambar**
reopens #59, rebased.

### CODE REVIEW

**Comment by fsmanuel** [comment]
@xcambar can we make it backwards compatible? I'll add some comments in the code.

File: addon/components/bread-crumbs.js [hunk 1]
```diff
@@ -23,7 +23,6 @@ export default Component.extend({
   tagName: 'ol',
   linkable: true,
   reverse: false,
-  classNameBindings: ['breadCrumbClass'],
```
Comment (fsmanuel) [inline addon/components/bread-crumbs.js]: `classNameBindings: ['breadCrumbClasses'],`

File: addon/components/bread-crumbs.js [hunk 2]
```diff
@@ -42,20 +41,6 @@ export default Component.extend({
     }
   }).readOnly(),
 
-  breadCrumbClass: computed('outputStyle', {
```
Comment (fsmanuel) [inline addon/components/bread-crumbs.js]:
```js
// Deprecated: outputStyle will be removed in next major release
breadCrumbClasses: computed('outputStyle', 'breadCrumbClass', {
  // all the old logic
  return `${className} ${get(this, 'breadCrumbClass')}`
}
```

**Comment by xcambar** [comment]
I've updated with a deprecation message when outputStyle is used.

I didn't add the `breadCrumbClasses` property as your comments suggested, as using the standard `class` parameter did the same job considering that `outputStyle` was in the `classNamesBindings` list.

See the implementation for all the details.

**Comment by dguettler** [comment]
@xcamber IMO the `breadCrumbClasses` should stay to not break existing functionality in the case someone overrode it

**Comment by xcambar** [comment]
I don't understand. `breadCrumbClasses` is not in master, it's been introduced (to my knowledge) by your comments in this PR.

And assuming you're talking about `breadCrumbClass`, it's a readOnly property, therefore it can *not* be overridden.

**Comment by dguettler** [comment]
Sorry was replying from my phone - yes `breadCrumbClass`

Not by passing it as an attribute. However if you extended the `bread-crumbs` component you could override the `breadCrumbClass` computed property.
Maybe an edge case, but still something to consider IMO.

**Comment by xcambar** [comment]
I've updated by reenabling `breadCrumClass` as a `classNameBindings` (https://github.com/poteto/ember-crumbly/pull/97/files#diff-cbc0ba5e52790aa45a4b879d6be5128cR27).

I didn't attach any behaviour nor default value, which means:
* it leaves the doors open to backward-compatible extension
* it has no effect by default

As of this comment, the behaviour is backward compatible, which means:
* `outputStyle` is deprecated yet still functional
* `breadCrumbClass` can still be overridden by extension

**Comment by dguettler** [comment]
@xcambar that seems reasonable, @fsmanuel any opinion?
Only remaining issue are the failing tests. Thanks for the work on this!

**Comment by xcambar** [comment]
master was failing last time I checked. I'll rebase when it is fixed.

**Comment by dguettler** [comment]
@xcambar sorry, I just noticed one case (and I can't get the test to show it to work) where the changes are creating an issue.

Since now both `outputStyle` and `breadCrumbClass` are evaluated for the class name you end up with a component which has both the default `breadcrumb` and a custom class name.

I'd propose to only bind the `breadCrumbClass` restore the original computed property and add
```
  outputStyle: computed({
    get() {
      return;
    },
    set(_, v) {
      deprecate('outputStyle option will be deprecated in the next major release', false);
      if (v.toLowerCase() === 'foundation') {
        return 'breadcrumbs';
      }
      return 'breadcrumb';
    }
  }),
```
for the deprecation.

Thank you for all the work and sorry for the long back and forth.

**Comment by xcambar** [comment]
You seem to want a completely different PR altogether.
From the very beginning, this PR was meant to be backward-incompatible, not only from a JS perspective but also because it removes the CSS declarations used by the `breadcrumb`/`breadcrumbs` classes.

If you only want to deprecate `outputStyle` (keep the behaviour, warn of upcoming unavailability), see #98.
When you feel that removing `outputStyle` is ready fo prime-time, feel free to rebase and merge this PR.

**Comment by dguettler** [comment]
Not exactly, I'm ok removing it from the documentation to avoid confusion
etc.
However I do not want to break existing functionality as this would require
a major version bump.

**Pull request closed without merging**